src/libs/signMessage/signMessage.ts

```diff
@@ -559,3 +635,52 @@ export function adjustEntryPointAuthorization(signature: string): string {
   // we adjust the entry point authorization signature so we could execute a txn
   return wrapStandard(entryPointSig.substring(0, entryPointSig.length - 2))
 }
+
+// the hash the user needs to eth_sign in order for his EOA to turn smarter
+export function getAuthorizationHash(chainId: bigint, contractAddr: Hex, nonce: bigint): Hex {
+  return keccak256(
+    concat([
+      '0x05', // magic authrorization string
+      encodeRlp([
+        // zeros are empty bytes in rlp encoding
+        chainId !== 0n ? toBeHex(chainId) : '0x',
+        contractAddr,
+        // zeros are empty bytes in rlp encoding
+        nonce !== 0n ? toBeHex(nonce) : '0x'
+      ])
+    ])
+  ) as Hex
+}
+
+export function getVerifyMessageSignature(
+  signature: EIP7702Signature | string,
+  account: Account,
+  accountState: AccountOnchainState
+): Hex {
+  if (isHexString(signature)) {
+    return account.creation && !accountState.isDeployed
+      ? // https://eips.ethereum.org/EIPS/eip-6492
+        (wrapCounterfactualSign(signature, account.creation) as Hex)
+      : (signature as Hex)
+  }
+
+  const sig = signature as EIP7702Signature
+  // ethereum v is 27 or 28
+  const v = get7702SigV(sig)
+  return concat([sig.r, sig.s, v]) as Hex
+}
+
+export function getSavedSignature(
+  signature: EIP7702Signature | string,
+  account: Account,
+  accountState: AccountOnchainState
+): EIP7702Signature | Hex {
+  if (isHexString(signature)) {
+    return account.creation && !accountState.isDeployed
+      ? // https://eips.ethereum.org/EIPS/eip-6492
+        (wrapCounterfactualSign(signature, account.creation) as Hex)
+      : (signature as Hex)
+  }
```

Comment:
6492 code is duplicated here and into getVerifyMessageSignature, we could reuse it